native/exe/ServiceControlLaunch/ServiceControlLaunch.cpp

```diff
@@ -104,10 +101,12 @@ void popupExceptionError(JNIEnv* env, jthrowable thrower, char* errTitle)
 	jmethodID toStr = env->GetMethodID(env->GetObjectClass(thrower), "toString", "()Ljava/lang/String;");
 	jstring throwStr = (jstring)env->CallObjectMethod(thrower, toStr);
 	const char* cThrowStr = env->GetStringUTFChars(throwStr, 0);
-	char *errStr = (char*) malloc(env->GetStringLength(throwStr) + 64);
-	sprintf(errStr, "An exception occured in Java:\n%s", cThrowStr);
+	size_t errStrlen = env->GetStringLength(throwStr) + 64;
+	char *errStr = (char*) malloc(errStrlen);
+	sprintf_s(errStr, errStrlen, "An exception occured in Java:\n%s", cThrowStr);

```

Comment:
Switched over to the Visual Studio 2015 (v140) Platform Toolset and ran code analysis and it complained about the sprintf_s with warning C6387: 'errStr' could be '0':  this does not adhere to the specification for the function 'sprintf_s'.
I can make the warning go away with
```
	if (errStr !=nullptr)
		sprintf_s(errStr, errStrlen, "An exception occured in Java:\n%s", cThrowStr);
```
but isn't errStr going to be NULL until the sprintf_s ?

Reply:
errStr is a pointer and is NULL until it points to a valid heap memory address (assigned via the 'new' operator).  The sprintf_s function does not allocate the memory.  The warning (and your valid fix) is to prevent passing that NULL pointer into sprintf_s in the (very rare) event that 'new' cannot allocate memory.

Reply:
Actually, you should wrap the sprintf_s, errMsg and delete lines all inside the nullptr check since none are valid with NULL.

Reply:
Done